R/to_integer.R

```diff
@@ -301,7 +307,26 @@ to_integer.gs_design <- function(x, sample_size = TRUE, round_up_final = TRUE, .
     }
 
     # Updated sample size to integer and enroll rates
-    sample_size_new <- (ceiling(x$analysis$n[n_analysis] / multiply_factor) * multiply_factor) %>% as.integer()
+    # if the randomization ratio is a whole number, round the sample size as a multiplier of ratio + 1
+    if(is_wholenumber(ratio)) {
+      ss <- x$analysis$n[n_analysis] / multiply_factor
+
+      if (round_up_final) {
+        sample_size_new <- ceiling(ss) * multiply_factor
+      } else {
+        sample_size_new <- round(ss, 0) * multiply_factor
+      }
+    # if the randomization ratio is NOT a whole number, just round it
+    } else {
+      if (round_up_final) {
+        sample_size_new <- ceiling(x$analysis$n[n_analysis])
+      } else {
+        sample_size_new <- round(x$analysis$n[n_analysis], 0)
+      }
+    }
+
+    sample_size_new <- as.integer(sample_size_new)
```

Comment:
@LittleBeannie This is the change that I was hoping you to consider. The rest of my suggestions are not important.

Reply:
Thank you so much for the suggestion, @yihui. Following, please find my counter example, where the shorten if-else may not work properly. 
![image](https://github.com/user-attachments/assets/cede2c81-1133-4db7-aa0f-1c62ce815afe)

Reply:
Thank you so much for the proposal, @yihui! Following, please find a counter example where the shorten if-else may not work properly. 
![image](https://github.com/user-attachments/assets/c3a06283-be1b-4527-9776-1402ebeb29ba)

Reply:
Thanks for the explanation! Perhaps I didn't make it clear enough---this change needs to happen together with the previous change on line 281 (which you have accepted), where we set `ratio` to 0 when it's not a whole number. If `ratio = 1.5`, `multiply_factor = ratio + 1 = 0 + 1 = 1`, therefore you get `as.integer([ceiling|round](100.1/1) * 1)`, i.e., 101 or 100.

Reply:
Hi @yihui, thank you for your guidance on this. After some further consideration, I would prefer not to set `ratio <- 0` when `ratio` is a non-integer. The reasoning is outlined below.

As indicated in https://github.com/Merck/gsDesign2/blob/487-sync-with-gsdesign-when-integering-sample-size/R/to_integer.R#L351, the `ratio` is used not only for determining the integer sample size but also for calculating statistical information (through `gs_info_ahr` or `gs_info_wlr`). If I set the `ratio` to 0, it would result in inaccurate statistical information.

Reply:
With the two further changes, I think you'll get what you want without the nested `if` statements. Essentially, what you need is `multiply_factor = ratio + 1` when `ratio` is a whole number, and `multiply_factor = 1` otherwise. Previously I was suggesting setting `ratio = 0` to achieve this, but it can also be achieved without changing `ratio` but setting `multiply_factor` directly.

Reply:
Thank you, @yihui! I like your suggestion to simplify the code. While I still prefer my through straightforward but “clumsy” if-else statements because they clearly outline all possible statistical scenarios, which I believe makes future maintenance easier... After this function is fully matured, let's  revisit them and make it more coding perfect.

Reply:
Sure. My principle is always that when in doubt, let the package maintainer decide. Even if I feel the code is suboptimal, I completely respect your decision, since you are the one who carry the package on the back, and your comfort should be of higher priority than ours.